app/models/work.rb

```diff
@@ -161,6 +147,9 @@ def valid_to_draft
   def valid_to_submit
     valid_to_draft
     validate_metadata
+    if state == "approved"
```

Comment:
I believe that update_ark_information and publish_doi should likely be in the same after: method.

The way this is written it will happen during validation even if the item is not valid.  [Maybe](https://github.com/pulibrary/pdc_describe/blob/8ea4e5d67be70c2bc4b966d934821a447b147eb4/app/models/work.rb#L31) becomes 
```
transitions from: :awaiting_approval, to: :approved, guard: :valid_to_submit, after: publish
...
def publish
   publish_doi
   update_ark_information
end
```

Reply:
Done. Added the `after` trigger and move the code to a new `publish` method